src/nlp/arrow_dataset.py

```diff
@@ -1522,7 +1560,7 @@ def train_test_split(
             )
 
         # Check if we've already cached this computation (indexed by a hash)
-        if self._data_files:
+        if self._data_files and (seed is not None or generator is not None):
```

Comment:
same here

instead of testing for seed/generator being None or not, simply move 
```python
            if generator is None:
                generator = np.random.default_rng(seed)
```
right before this line

Reply:
Great. I'm assuming it's ok to move this out of the if statement that it was in before? Ok to create the generator and just not use it when not shuffling?

Reply:
Yes it's fine